comp/trace/config/setup.go

```diff
@@ -238,61 +257,62 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 		log.Debug("Found APM feature flags: %v", c.Features)
 	}
 
-	if k := "apm_config.ignore_resources"; coreconfig.Datadog.IsSet(k) {
-		c.Ignore["resource"] = coreconfig.Datadog.GetStringSlice(k)
+	if k := "apm_config.ignore_resources"; core.IsSet(k) {
+		c.Ignore["resource"] = core.GetStringSlice(k)
 	}
-	if k := "apm_config.max_payload_size"; coreconfig.Datadog.IsSet(k) {
-		c.MaxRequestBytes = coreconfig.Datadog.GetInt64(k)
+	if k := "apm_config.max_payload_size"; core.IsSet(k) {
+		c.MaxRequestBytes = core.GetInt64(k)
 	}
-	if k := "apm_config.replace_tags"; coreconfig.Datadog.IsSet(k) {
+	if k := "apm_config.replace_tags"; core.IsSet(k) {
 		rt := make([]*config.ReplaceRule, 0)
 		if err := coreconfig.Datadog.UnmarshalKey(k, &rt); err != nil {
 			log.Errorf("Bad format for %q it should be of the form '[{\"name\": \"tag_name\",\"pattern\":\"pattern\",\"repl\":\"replace_str\"}]', error: %v", "apm_config.replace_tags", err)
 		} else {
 			err := compileReplaceRules(rt)
 			if err != nil {
-				osutil.Exitf("replace_tags: %s", err)
+				// osutil.Exitf("replace_tags: %s", err)
+				return fmt.Errorf("replace_tags: %s", err)
```

Comment:
Should it be `osutil.Exitf` or returning the error? Can you delete whichever it shouldn't be?

Reply:
We should be returning an error, `uber-go/fx` should be able to handle the error bubbling up, and we can get rid of this anti-pattern of just dying like that with no cleanup.